encoding/json/encode.go

```diff
@@ -618,10 +627,14 @@ func prepareComposite(kind, id string, fieldTypes []cadence.Field, fields []cade
 	compositeFields := make([]jsonCompositeField, len(fields))
 
 	for i, value := range fields {
-		fieldType := fieldTypes[i]
+		var name string
+		if i < len(fieldTypes) {
+			fieldType := fieldTypes[i]
+			name = fieldType.Identifier
+		}
```

Comment:
We already check this at the top of the function, so wouldn't need to do it here again?

Reply:
I'll add comments to clarify

Reply:
Explained it in [a9b2f77](https://github.com/onflow/cadence/pull/2517/commits/a9b2f77ed8350778c742eb9a69cdf6118f028e83)